akka-http/src/main/scala/akka/http/javadsl/server/directives/WebSocketDirectives.scala

```diff
@@ -44,17 +44,16 @@ abstract class WebSocketDirectives extends SecurityDirectives {
    * Handles WebSocket requests with the given handler and rejects other requests with an
    * [[ExpectedWebSocketRequestRejection]].
    */
-  def handleWebSocketMessages[T](handler: Flow[Message, Message, T]): Route = RouteAdapter {
+  def handleWebSocketMessages(handler: Flow[Message, Message, NotUsed]): Route = RouteAdapter {
```

Comment:
Another option could perhaps be `handleWebSocketMessages[M](handler: Flow[Message, Message, M]): Route`. That way you'd never have to adapt the materialized value to some specific type?